react/features/prejoin/actions.js

```diff
@@ -0,0 +1,354 @@
+    SET_PREJOIN_AUDIO_MUTED,
+    SET_PREJOIN_DEVICE_ERRORS,
+    SET_PREJOIN_NAME,
+    SET_PREJOIN_PAGE_VISIBILITY,
+    SET_PREJOIN_VIDEO_MUTED
+} from './actionTypes';
+import { createLocalTrack } from '../base/lib-jitsi-meet';
+import { getUserSelectedCameraDeviceId } from '../base/settings';
+import {
+    isContentSharingActive,
+    getAudioTrack,
+    getVideoTrack
+} from './functions';
+import logger from './logger';
+
+/**
+ * Action used to add an audio track to the store.
+ *
+ * @param {Object} value - The track to be added.
+ * @returns {Object}
+ */
+export function addPrejoinAudioTrack(value: Object) {
+    return {
+        type: ADD_PREJOIN_AUDIO_TRACK,
+        value
+    };
+}
+
+/**
+ * Action used to add a video track to the store.
+ *
+ * @param {Object} value - The track to be added.
+ * @returns {Object}
+ */
+export function addPrejoinVideoTrack(value: Object) {
+    return {
+        type: ADD_PREJOIN_VIDEO_TRACK,
+        value
+    };
+}
+
+/**
+ * Action used to add a content sharing track to the store.
+ *
+ * @param {Object} value - The track to be added.
+ * @returns {Object}
+ */
+export function addPrejoinContentSharingTrack(value: Object) {
+    return {
+        type: ADD_PREJOIN_CONTENT_SHARING_TRACK,
+        value
+    };
+}
+
+/**
+ * Creates and adds a new video track to the store.
+ * Used when there is already a screen sharing(video) track created.
+ *
+ * @returns {Function}
+ */
+function createAndAddPrejoinVideoTrack() {
+    return async function(dispatch: Function, getState: Function) {
+        try {
+            const deviceId = getUserSelectedCameraDeviceId(getState());
+            const track = await createLocalTrack('video', deviceId);
+
+            dispatch(addPrejoinVideoTrack(track));
+
+            return track;
+        } catch (err) {
+            dispatch(setDeviceStatusWarning('prejoin.videoTrackError'));
+            logger.log('Could not create local video track', err);
+        }
+    };
+}
+
+
+/**
+ * Adds all the newly created tracks to store on init.
+ *
+ * @param {Object[]} tracks - The newly created tracks.
+ * @param {Object} errors - The errors from creating the tracks.
+ * @param {boolean} startScreenSharing - If there is an initial screen sharing track.
+ *
+ * @returns {Function}
+ */
+export function initPrejoin(tracks: Object[], errors: Object, { startScreenSharing }: { startScreenSharing: boolean}) {
+    return async function(dispatch: Function) {
+        const audioTrack = tracks.find(t => t.isAudioTrack());
+        const videoTrack = tracks.find(t => t.isVideoTrack());
+
+        dispatch(setPrejoinDeviceErrors(errors));
+
+        if (audioTrack) {
+            dispatch(addPrejoinAudioTrack(audioTrack));
+        } else {
+            dispatch(setAudioDisabled());
+        }
+
+        if (startScreenSharing) {
+            dispatch(addPrejoinContentSharingTrack(videoTrack));
+            dispatch(createAndAddPrejoinVideoTrack());
```

Comment:
If there is screen sharing track created because there are initial options for it, this is the default video track. 

I have to create an additional track for the camera. The newly created track is used only as long as the user configures the settings and then disposed. The CS track is passed as the default video track, but the settings for video are being persisted, so when camera is active, should pick up those settings.

...Or maybe I'm missing something here 🤔

Reply:
If the settings say that the call should start with CS, then there is no need to create a camera track. It will be created later, when the user unmutes, if they choose to.

I'd say we should disable the camera button if the call starts with CS.

Reply:
I have followed your advice. The camera button is disabled now if the meeting starts with CS.